### PR TITLE
fix(tick): move tickEpoch to separate function

### PR DESCRIPTION
**Pull request by dtfiedler**
We will add unit tests to make sure this happens correctly

### CODE REVIEW

File: src/main.lua [hunk 1]
```diff
@@ -1574,36 +1576,11 @@ end)
...
-		local newEpoch = epochs.createEpoch(timestamp, tonumber(blockHeight), hashchain)
-		return {
-			maybeEpoch = newEpoch,
-			maybeDemandFactor = demandFactor,
-			pruneGatewaysResult = pruneGatewaysResult,
-		}
-	end
 
 	local msgId = msg.Id
+	local blockHeight = tonumber(msg["Block-Height"])
```
Comment (arielmelendez) [inline src/main.lua:1582]: Should be a tag?

Comment (arielmelendez) [reply src/main.lua:1582]: And also already tonumbered?

File: src/main.lua [hunk 2]
```diff
@@ -1574,36 +1576,11 @@ end)
...
+	local hashchain = msg["Hash-Chain"]
```
Comment (arielmelendez) [inline src/main.lua:1583]: tag?

File: src/main.lua (hunk 1 above)
Comment (dtfiedler) [reply src/main.lua:1582]: This is a top level attribute on all messages, similar to timestamp

**Review by fedellen — DISMISSED** [review]

**Pull request merged**